crates/cubecl-runtime/src/server.rs

```diff
@@ -136,6 +174,43 @@ pub struct Binding {
     pub offset_end: Option<u64>,
 }
 
+/// Binding of a grid constant to execute a kernel.
+#[derive(new, Debug, Clone)]
+pub enum ConstBinding {
+    /// A tensor map used for TMA loading ops
+    TensorMap {
+        /// The binding for the backing tensor
+        binding: Binding,
+        /// The tensormap metadata
+        map: TensorMap,
+    },
+}
+
+/// TensorMap metadata for the opaque proxy used in TMA copies
+#[derive(Debug, Clone)]
+pub struct TensorMap {
+    /// Tensormap format (tiled or im2col)
+    pub format: TensorMapFormat,
+    /// Rank of the backing tensor
+    pub rank: usize,
+    /// Shape of the backing tensor
+    pub shape: Vec<u64>,
+    /// Strides of the backing tensor
+    pub strides: Vec<u64>,
+    /// Element stride, usually 1 but may be 2 for complex tensors
+    pub elem_stride: Vec<u32>,
```

Comment:
Why is this a vec?

Reply:
Because it depends on dimensionality and const generics don't really work here.